lunabot_embedded/firmware/teensy_main/lib/lunabot_drivers/interfaces.cpp

```diff
@@ -252,3 +241,99 @@ void M5Stack_UWB_Trncvr::transfer() {
     }
   }
 }
+
+long KillSwitchRelay::kill_time;
+bool KillSwitchRelay::dead;
+
+void KillSwitchRelay::init() { 
+  pinMode(kill_pin, OUTPUT); 
+  reset();
+  KillSwitchRelay::kill_time = millis();
+}
+
+void KillSwitchRelay::reset() { 
+  digitalWrite(kill_pin, HIGH); 
+  KillSwitchRelay::dead = false;
+}
+
+void KillSwitchRelay::kill() { 
+  digitalWrite(kill_pin, LOW);
+  KillSwitchRelay::kill_time = millis();
+  KillSwitchRelay::dead = true;
+}
+
+
+
+void KillSwitchRelay::kill_motor(int id, RobotEffort &effort) {
+  switch (id) {
+    case 0:
+      effort.excavate = 0;
+      break;
+    case 1:
+      effort.deposit = 0;
+      break;
+    case 2:
+      effort.left_drive = 0;
+      break;
+    case 3:
+      effort.right_drive = 0;
+      break;
+    default:
+      return;
+  }
+}
+
+//exc, dep, drive_L, drive_R
+volatile float KillSwitchRelay::cutoff_buffer[4] = {0};
+volatile float KillSwitchRelay::kill_buffer[4] = {0};
+volatile bool KillSwitchRelay::is_dead[4] = {false};
+
+void KillSwitchRelay::logic(RobotEffort &effort) {
+  if (KillSwitchRelay::dead && millis() - KillSwitchRelay::kill_time >= relay_dead_time) {
+    reset();
+  } 
```

Comment:
I personally think we should remove this. if motors really get killed, atleast for now since we have untested hardware, we should be very careful and not allow automatic resets) can easily change in the future. (right now its 2 sec which actually is pretty fast)

Reply:
What exactly do you mean by remove it? We have no other way of turning the motors back on right now. I can make the time the relay stays dead longer, but it seems bad to turn the motors off entirely with no way of getting them back.

Reply:
Okay. Then make the reset time like 1min? 2 seconds is too short with untested hardware.